### PR TITLE
Move gc e2e to one namespace per spec

### PR DESCRIPTION
**Pull request by perdasilva**
Signed-off-by: perdasilva <perdasilva@redhat.com>

**Description of the change:**
This PR moves the gc e2e suite to use one namespace per spec

**Motivation for the change:**
e2e stability

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/doc`
- [ ] Commit messages sensible and descriptive
- [ ] Tests marked as `[FLAKE]` are truly flaky
- [ ] Tests that remove the `[FLAKE]` tag are no longer flaky

### CODE REVIEW

File: test/e2e/gc_e2e_test.go [hunk 1]
```diff
@@ -27,15 +28,25 @@ var _ = Describe("Garbage collection for dependent resources", func() {
...
 		kubeClient     operatorclient.ClientInterface
 		operatorClient versioned.Interface
+		ns             corev1.Namespace
 	)
 
 	BeforeEach(func() {
 		kubeClient = ctx.Ctx().KubeClient()
 		operatorClient = ctx.Ctx().OperatorClient()
+
+		namespaceName := genName("catsrc-e2e-")
```
[the remark this answers is not in the document]
Comment (timflannagan) [reply test/e2e/gc_e2e_test.go:38]: +1

**Comment by timflannagan** [comment]
Pulling this down locally now but this looks good to go outside of changing that generated namespace prefix.

**Comment by timflannagan** [comment]
Around ran this locally a couple of times and didn't encounter any cluster-wide (or any hanging test generated namespaces) resource leakage.

File: test/e2e/gc_e2e_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (perdasilva) [reply test/e2e/gc_e2e_test.go]: Fixed ^^

**Review by timflannagan — APPROVED** [review]
/lgtm

**Pull request merged**